src/python/tools/preprocess/converters/writers/spark_writer.py

```diff
@@ -109,11 +111,18 @@ def write_partitioned_df_to_csv(partition_triples, num_partitions, output_filena
 
 
 class SparkWriter(object):
-    def __init__(self, spark, output_dir, partitioned_evaluation):
+    def __init__(self, spark, output_dir, output_to_s3, partitioned_evaluation):
         super().__init__()
 
         self.spark = spark
         self.output_dir = output_dir
+        self.output_to_s3 = output_to_s3
+        if self.output_to_s3:
+            self.s3_bucket = os.getenv("S3_BUCKET")
```

Comment:
Can we write a readme document or have some documentation that lists all the env variables that a user is required to set

Reply:
sure, will check in a README too.